### PR TITLE
Add validate helper function

### PR DESCRIPTION
**Pull request by willthames**

### CODE REVIEW

**Comment by fabianvf** [comment]
@willthames 

Can you move this from openshift/helper to openshift/dynamic? (The helper code is all focused on the generated approach, and will be deprecated alongside it). I would just add validate as a function on the DynamicClient (which would allow us to automatically determine the Kubernetes version and eliminate the client argument), and raise an exception if that function is called but the module is not installed. I wouldn't call it from create/delete/path/replace, instead I would expect the user to do something like:

```python
client = DynamicClient()
v1_pods = client.resources.get(api_version='v1', kind='Pod')
definition = {'apiVersion': 'v1', 'kind': 'Pod', ...}

client.validate(definition)
v1_pods.create(definition)
```

Does that approach seem ok to you?

File: openshift/helper/validate.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    warnings = list()
+    try:
+        if version is None:
+            if client:
+                version = client.version['kubernetes']['gitVersion']
+            else:
+                version = kubernetes_validate.latest_version()
+        kubernetes_validate.validate(resource, version, strict)
+    except kubernetes_validate.utils.ValidationError as e:
+        errors.append("resource validation error at %s: %s" % ('.'.join([str(item) for item in e.path]), e.message))
```
Comment (fabianvf) [inline openshift/helper/validate.py]: I think `e.message` is making linting for python3 fail, I guess it was removed. The message is

```
B306 `BaseException.message` has been deprecated as of Python 2.6 and is removed in Python 3. 
Use `str(e)` to access the user-readable message. 
Use `e.args` to access arguments passed to the exception.
```

**Comment by willthames** [comment]
@fabianvf thanks for your feedback as always, that's great - I'll implement those changes in the next few days

File: openshift/helper/validate.py (hunk 1 above)
Comment (willthames) [reply openshift/helper/validate.py]: @fabianvf - `e.message` was removed from BaseException, sure. But ValidationError sets `message` explicitly.

https://python-jsonschema.readthedocs.io/en/latest/errors/#jsonschema.exceptions.ValidationError

If I test almost identical code in python3 from kubernetes-validate codebase, it works fine:
```
PYTHONPATH=src python3 src/kubernetes_validate/__main__.py --strict examples/kuard-extra-property.yaml 
ERROR examples/kuard-extra-property.yaml did not validate against version 1.11.0: spec.selector: Additional properties are not allowed ('unwanted' was unexpected)
```

File: openshift/dynamic/client.py [hunk 2]
```diff
@@ -256,6 +263,38 @@ def request(self, method, path, body=None, **params):
         )
 
 
+    def validate(self, resource, version=None, strict=False):
```
Comment (fabianvf) [inline openshift/dynamic/client.py]: would you mind renaming `resource` here? In the methods above `resource` is used to refer to the class of resource, not an individual instance. I think `definition` would be better.

File: openshift/dynamic/client.py [hunk 3]
```diff
@@ -256,6 +263,38 @@ def request(self, method, path, body=None, **params):
...
+        """validate checks a kubernetes resource definition
+
+        Args:
+            resource (dict): resource definition
+            client (object): openshift dynamic client used to obtain version
```
Comment (fabianvf) [inline openshift/dynamic/client.py]: I think this is no longer accurate

**Comment by willthames** [comment]
@fabianvf I've pushed a fix for that

**Pull request merged**